File: app/src/main/java/md/fusionworks/lifehack/ui/sales/SaleCategorySpinnerAdapter.java
package md.fusionworks.lifehack.ui.sales;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;
import java.util.ArrayList;
import java.util.List;
import md.fusionworks.lifehack.ui.sales.model.SaleCategoryModel;
import md.fusionworks.lifehack.util.Constant;

/**
 * Created by ungvas on 10/30/15.
 */

public class SaleCategorySpinnerAdapter extends BaseAdapter {

  private Context context;
  private List<SaleCategoryModel> items = new ArrayList<>();
  private String language;

  public SaleCategorySpinnerAdapter(Context context, List<SaleCategoryModel> items,
      String language) {
    this.context = context;
    this.items = items;
    this.language = language;
  }

  @Override public int getCount() {
    return items.size();
  }

  @Override public Object getItem(int position) {
    return items.get(position);
  }

  @Override public long getItemId(int position) {
    return items.get(position).id;
  }

  private String getTitle(int position) {
    return position >= 0 && position < items.size() ? (language.equals(Constant.LANG_RU))
        ? items.get(position).nameRu : items.get(position).nameRo : "";
  }

  @Override public View getDropDownView(int position, View view, ViewGroup parent) {
    view = ((Activity) context).getLayoutInflater()
        .inflate(android.R.layout.simple_spinner_dropdown_item, parent, false);

    TextView normalTextView = (TextView) view.findViewById(android.R.id.text1);
    normalTextView.setText(getTitle(position));

    return view;
  }

  @Override public View getView(int position, View view, ViewGroup parent) {
    view = ((Activity) context).getLayoutInflater()
        .inflate(android.R.layout.simple_spinner_item, parent, false);

    TextView textView = (TextView) view.findViewById(android.R.id.text1);
    textView.setText(getTitle(position));
    return view;
  }
}

File: app/src/main/java/md/fusionworks/lifehack/ui/sales/SaleCategorySpinnerAdapter.kt
package md.fusionworks.lifehack.ui.sales

import android.app.Activity
import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import md.fusionworks.lifehack.ui.sales.model.SaleCategoryModel
import md.fusionworks.lifehack.util.Constant

/**
 * Created by ungvas on 10/30/15.
 */

class SaleCategorySpinnerAdapter(private val context: Context, private val itemList: List<SaleCategoryModel>,
    private val language: String) : BaseAdapter() {

  override fun getCount(): Int {
    return itemList.size
  }

  override fun getItem(position: Int): Any {
    return itemList[position]
  }

  override fun getItemId(position: Int): Long {
    return itemList[position].id.toLong()
  }

  private fun getTitle(position: Int): String {
    return if (position >= 0 && position < itemList.size) if (language == Constant.LANG_RU) itemList[position].nameRu
    else itemList[position].nameRo
    else ""
  }

  override fun getDropDownView(position: Int, view: View?, parent: ViewGroup): View {
    val  nview = (context as Activity).layoutInflater.inflate(
        android.R.layout.simple_spinner_dropdown_item, parent, false)

    val normalTextView = nview.findViewById(android.R.id.text1) as TextView
    normalTextView.text = getTitle(position)

    return nview
  }

  override fun getView(position: Int, view: View?, parent: ViewGroup): View {
    val nview = (context as Activity).layoutInflater.inflate(
        android.R.layout.simple_spinner_item,
        parent, false)

    val textView = nview.findViewById(android.R.id.text1) as TextView
    textView.text = getTitle(position)
    return nview
  }
}
